libsoftwarecontainer/src/gateway/devicenodegateway.cpp

```diff
@@ -86,18 +91,32 @@ bool DeviceNodeGateway::activateGateway()
             parseInt(dev.mode.c_str(), &mode);
 
             // mknod dev.name c dev.major dev.minor
-            if (mknod(dev.name.c_str(), S_IFCHR | mode, makedev(majorVersion, minorVersion)) != 0) {
+            pid_t pid = INVALID_PID;
+            getContainer()->executeInContainer([&] () {
+                  return mknod(dev.name.c_str(), S_IFCHR | mode,
+                               makedev(majorVersion, minorVersion));
+                  }, &pid);
```

Comment:
To me the indentation of the bracket here looks misaligned. But it looks consistent with the rest so perhaps this is how we want it?